src/core/action-creator.ts

```diff
@@ -51,7 +51,21 @@ export function createActions(flux: FluxCapacitor) {
                 func: ({ query }) => !('query' in search) || !!query || query === null,
                 msg: 'search term is empty'
               }, {
-                func: ({ query }, state) => query !== Selectors.query(state) || query === null,
+                  // currently assume that search will only ever have up to one refinement
+                  func: ({ query, value, navigationId }, state) => {
```

Comment:
there are actually a few more options for the parameters to this function, `low`, `high`, `range` and `refinementId`